packages/flutter/lib/src/material/menu_anchor.dart

```diff
@@ -2464,6 +2489,397 @@ class _MenuDirectionalFocusAction extends DirectionalFocusAction {
+typedef MenuAcceleratorChildBuilder = Widget Function(
+  BuildContext context,
+  String label,
+  int index,
+);
+
+/// A widget that draws the label text for a menu item (typically a
+/// [MenuItemButton] or [SubmenuButton]) and renders its child with information
+/// about the currently active keyboard accelerator.
+///
+/// On platforms other than macOS and iOS, this widget listens for the Alt key
+/// to be pressed, and when it is down, will update the label by calling the
+/// builder again with the position of the accelerator in the label string.
+/// While the Alt key is pressed, it registers a shortcut with the
+/// [ShortcutRegistry] mapped to a [VoidCallbackIntent] containing the callback
+/// defined by the nearest [MenuAcceleratorCallbackBinding].
+///
+/// Because the accelerators are registered with the [ShortcutRegistry], any
+/// other shortcuts in the widget tree between the [primaryFocus] and the
+/// [ShortcutRegistry] that define Alt-based shortcuts using the same keys will
+/// take precedence over the accelerators.
+///
+/// Because accelerators aren't used on macOS and iOS, the label ignores the Alt
+/// key on those platforms, and the [builder] is always given -1 as an
+/// accelerator index. Accelerator labels are still stripped of their
+/// accelerator markers.
+///
+/// The built-in menu items [MenuItemButton] and [SubmenuButton] already provide
+/// the appropriate [MenuAcceleratorCallbackBinding], so unless you are creating
+/// your own custom menu item type that takes a [MenuAcceleratorLabel], it is
+/// not necessary to provide one.
+///
+/// {@template flutter.material.MenuAcceleratorLabel.accelerator_sample}
+/// {@tool dartpad} This example shows a [MenuBar] that handles keyboard
+/// accelerators using [MenuAcceleratorLabel]. To use the accelerators, press
+/// the Alt key to see which letters are underlined in the menu bar, and then
+/// press the appropriate letter. Accelerators are not supported on macOS or iOS
+/// since those platforms don't support them natively, so this demo will only
+/// show a regular Material menu bar on those platforms.
+///
+/// ** See code in
+/// examples/api/lib/material/menu_anchor/menu_accelerator_label.0.dart **
+/// {@end-tool}
+/// {@endtemplate}
+class MenuAcceleratorLabel extends StatefulWidget {
+  /// Creates a const [MenuAcceleratorLabel].
+  ///
+  /// The [label] parameter is required.
+  const MenuAcceleratorLabel(
+    this.label, {
+    super.key,
+    this.builder = defaultLabelBuilder,
+  });
+
+  /// The label string that should be displayed.
+  ///
+  /// The label string provides the label text, as well as the possible
+  /// characters which could be used as accelerators in the menu system.
+  ///
+  /// {@template flutter.material.menu_anchor.menu_accelerator_label.label}
+  /// To indicate which letters in the label are to be used as accelerators, add
+  /// a "&" character before the character in the string. If more than one
```

Comment:
Super nit: "a" => "an"?

Reply:
Looks like you do say "an &" below as well.

Reply:
Yes, it should be "an".